src/app/_components/forms/AuthForm.tsx

```diff
@@ -0,0 +1,157 @@
+    setNext(params.get("next"));
+  }, []);
+
+  const form: UseFormReturn<T> = useForm({
+    resolver: zodResolver(schema),
+    defaultValues: defaultValues as DefaultValues<T>,
+  });
+
+  const handleSubmit: SubmitHandler<T> = async (data) => {
+    console.log("values", data);
+  };
+
+  return (
+    <>
+      <Form {...form}>
+        <form onSubmit={form.handleSubmit(handleSubmit)} className="auth-form">
+          <h1 className="form-title">
+            {isSignIn ? "Inicia sesión en tu cuenta" : "Registrarse"}
+          </h1>
+          {Object.keys(defaultValues).map((field) => (
+            <FormField
+              key={field}
+              control={form.control}
+              name={field as Path<T>}
+              render={({ field }) => (
+                <FormItem>
+                  <div className="shad-form-item">
+                    <FormLabel className="shad-form-label capitalize">
+                      {FIELD_NAMES[field.name as keyof typeof FIELD_NAMES]}
+                    </FormLabel>
+                    <FormControl>
+                      <Input
+                        type={
+                          FIELD_TYPES[field.name as keyof typeof FIELD_TYPES]
+                        }
+                        placeholder={
+                          FIELD_PLACEHOLDERS[
+                            field.name as keyof typeof FIELD_PLACEHOLDERS
+                          ]
+                        }
+                        {...field}
+                        className="shad-input shad-no-focus"
+                      />
+                    </FormControl>
+                  </div>
+                  <FormMessage className="shad-form-message" />
+                </FormItem>
+              )}
+            />
+          ))}
+          {!isSignIn && (
+            <p className="mt-4 text-center text-sm text-neutral-500">
+              Al registrarte, aceptas nuestras{" "}
+              <Link href="#" className="text-[#141e3a] hover:underline">
+                Condiciones
+              </Link>{" "}
+              y nuestra{" "}
+              <Link href="#" className="text-[#141e3a] hover:underline">
+                Política de datos
+              </Link>
+              .
+            </p>
+          )}
+          <Button
+            type="submit"
+            className="form-submit-button"
+            disabled={isLoading}
+          >
+            {isSignIn ? "Ingresar" : "Registrarse"}
+            {isLoading && (
+              <Image
+                src="/assets/icons/loader.svg"
+                alt="loader"
+                width={24}
+                height={24}
+                className="ml-2 animate-spin"
+              />
+            )}
+          </Button>
+          {errorMessage && <p className="error-message">*{errorMessage}</p>}
+
+          <div className="flex w-full items-center">
+            <div className="flex-grow border-t border-neutral-300"></div>
```

Comment:
You can collapse the empty JSX tag into a self-closing tag:

```suggestion
            <div className="flex-grow border-t border-neutral-300" />
```